network/p2p/libp2pNode.go

```diff
@@ -0,0 +1,72 @@
+package p2p
+
+import (
+	"context"
+
+	kbucket "github.com/libp2p/go-libp2p-kbucket"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+	libp2pnet "github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+	"github.com/libp2p/go-libp2p/core/routing"
+
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+
+	"github.com/onflow/flow-go/network/channels"
+	"github.com/onflow/flow-go/network/p2p/unicast"
+)
+
+type LibP2PNode interface {
```

Comment:
Can we add docs as to why this interface is necessary in terms of creating different types of libp2p nodes, including corrupt ones used for testing?

Reply:
https://github.com/onflow/flow-go/pull/3276/commits/8af9933acf609cc1f2c7a90f64bcde77b16b5c3e